kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -53,13 +54,56 @@ grCollectionsPanel.controller('GrNodeCtrl', ['collections', function(collections
 
 }]);
 
+grCollectionsPanel.controller('GrAddToCollectionCtrl',
+    ['mediaApi', '$scope', function(mediaApi, $scope) {
+
+    const ctrl = this;
+
+    ctrl.saved = false;
+
+    ctrl.addImagesToCollection = imagesJson => {
+        const imageIds = imagesJson.map(imageJson => imageJson.id);
+        const promises = imageIds.map(id => mediaApi.find(id).then(image =>
```

Comment:
Is this looking up from the API again, rather than creating resources from the in-memory object? Not a big deal, just curious.